extensions/positron-python/python_files/posit/positron/positron_ipkernel.py

```diff
@@ -220,11 +221,48 @@ def connection_show(self, line: str) -> None:
 original_showwarning = warnings.showwarning
 
 
+class PositronDisplayFormatter(DisplayFormatter):
+    @traitlets.default("ipython_display_formatter")
+    def _default_formatter(self):
+        return PositronIPythonDisplayFormatter(parent=self)
+
+
+class PositronIPythonDisplayFormatter(IPythonDisplayFormatter):
+    print_method = traitlets.ObjectName("_ipython_display_")
+    _return_type = (type(None), bool)
+
+    @catch_format_error
+    def __call__(self, obj):
+        """Compute the format for an object."""
+        try:
+            if obj.__module__ == "plotnine.ggplot":
+                obj.draw(show=True)
+                return True
+        except AttributeError:
+            pass
+        if self.enabled:
+            # lookup registered printer
+            try:
+                printer = self.lookup(obj)
+            except KeyError:
+                pass
+            else:
+                printer(obj)
+                return True
+            # Finally look for special method names
+            method = dir2.get_real_method(obj, self.print_method)
+            if method is not None:
+                method()
+                return True
+        return True
+
+
 class PositronShell(ZMQInteractiveShell):
     kernel: PositronIPyKernel
     object_info_string_level: int
     magics_manager: MagicsManager
     display_pub: ZMQDisplayPublisher
+    display_formatter = traitlets.Instance(PositronDisplayFormatter)
```

Comment:
You might be able to remove the `# type: ignore` from L339 with:
```suggestion
    display_formatter: PositronDisplayFormatter = traitlets.Instance(PositronDisplayFormatter)  # type: ignore
```
We've done that elsewhere.